light/client_test.go

```diff
@@ -382,8 +363,19 @@ func TestClient_SkippingVerification(t *testing.T) {
 // start from a large light block to make sure that the pivot height doesn't select a height outside
 // the appropriate range
 func TestClientLargeBisectionVerification(t *testing.T) {
```

Comment:
I don't think this test is quite right. Not saying that the way you've converted it is wrong. I'm more blaming the person who initially wrote this test who is incidentally me. 

I don't know if you've figured out how light client verification (or skipping verification) works. If not, light clients start from a trusted height and verify a new height by finding a chain of blocks in between the two heights whereby 1/3+ of the validator set in the block we trust has signed in the new block we don't yet trust (the idea being that in that subset there is at least 1 honest validator). The way we work out the next height where there is sufficient overlap is a bisection technique where we keep halving the distance between the trusted height and whatever the next untrusted block is until there is enough overlap. Except we don't half it. We take a point 9/16ths in between. Peculiar right? The reason why it's 9/16th is because we cache blocks from previous verification attempts to reduce the amount of calls we make to providers. 

This test is designed to make sure that if we're trying to verify between two heights, we don't bisect a height that is **outside** the two heights. (Which might occur if we're calculating the 9/16ths part wrong). Hence we should be verifying from a height like 9000 to 9100 and make sure that no calls outside of that height are made by the light client. Does that make sense to you?

Reply:
I think that this does make sense. What it is making me wonder is why the test, as written, works. The only values that the mock node will return are for heights 5 and heights 100. The mock, as I've written it, will not return any values in between and will fail the test. Does that mean we are immediately trusting the block at height 100? Should we change that?

Reply:
Yeah because you set the `valVariation` field to 0, the validator sets don't change at all so the light client can verify directly from 5 to 100

Reply:
updated to use the range from 200-300, looking for block 256 for bisection.